modular_skyrat/modules/pet_owner/pet_owner.dm

```diff
@@ -0,0 +1,102 @@
+	mail_goodies = list(
+		/obj/item/clothing/neck/petcollar
+	)
+	var/pet_type = NONE
+
+/datum/quirk_constant_data/pet_owner
+	associated_typepath = /datum/quirk/item_quirk/pet_owner
+	customization_options = list(/datum/preference/choiced/pet_owner)
+
+/datum/quirk/item_quirk/pet_owner/add_unique(client/client_source)
+	var/desired_pet = client_source?.prefs.read_preference(/datum/preference/choiced/pet_owner) || "Random"
+
+	if(desired_pet != "Random")
+		pet_type = GLOB.possible_player_pet[desired_pet]
+
+	if(pet_type == NONE) // Pet not set, we're picking one for them.
+		pet_type = pick(flatten_list(GLOB.possible_player_pet))
+
+	var/obj/item/pet_carrier/carrier = new /obj/item/pet_carrier(get_turf(quirk_holder))
+	var/mob/living/simple_animal/pet/pet = new pet_type(carrier)
+	carrier.add_occupant(pet)
+	give_item_to_holder(
+		carrier,
+		list(
+			LOCATION_HANDS = ITEM_SLOT_HANDS
+		),
+		flavour_text = "Looks tightly packed - you might not be able to put the pet back in once they're out.",
+	)
+
+/datum/preference/choiced/pet_owner
+	category = PREFERENCE_CATEGORY_MANUALLY_RENDERED
+	savefile_key = "pet_owner"
+	savefile_identifier = PREFERENCE_CHARACTER
+	can_randomize = FALSE
+
+GLOBAL_LIST_INIT(possible_player_pet, list(
+	"Axolotl" = /mob/living/basic/axolotl,
+	"Baby Penguin" = /mob/living/basic/pet/penguin/baby/permanent,
+	"Bat" = /mob/living/basic/bat,
+	"Bread cat" = /mob/living/simple_animal/pet/cat/breadcat,
+	"Bull terrier" = /mob/living/basic/pet/dog/bullterrier,
+	"Butterfly" = /mob/living/basic/butterfly,
+	"Cake cat" = /mob/living/simple_animal/pet/cat/cak,
+	"Cat" = /mob/living/simple_animal/pet/cat,
+	"Chick" = /mob/living/basic/chick/permanent,
+	"Chicken" = /mob/living/basic/chicken,
+	"Chinchilla (dark)" = /mob/living/basic/pet/chinchilla/black,
+	"Chinchilla (white)" = /mob/living/basic/pet/chinchilla/white,
+	"Corgi" = /mob/living/basic/pet/dog/corgi,
+	"Corgi puppy" = /mob/living/basic/pet/dog/corgi/puppy,
+	"Cockroach" = /mob/living/basic/cockroach,
+	"Crab" = /mob/living/basic/crab,
+	"Deer" = /mob/living/basic/deer,
+	"Dobermann" = /mob/living/basic/pet/dog/dobermann,
+	"Fox" = /mob/living/basic/pet/fox/docile,
+	"Frog" = /mob/living/basic/frog,
+	"Giant ant" = /mob/living/basic/ant,
+	"Kitten" = /mob/living/simple_animal/pet/cat/kitten,
+	"Kiwi" = /mob/living/basic/kiwi,
+	"Mothroach" = /mob/living/basic/mothroach,
+	"Mouse (white)" = /mob/living/basic/mouse/white,
+	"Mouse (gray)" = /mob/living/basic/mouse/gray,
+	"Mouse (brown)" = /mob/living/basic/mouse/brown,
+	"Parrot" = /mob/living/basic/parrot,
+	"Penguin" = /mob/living/basic/pet/penguin/emperor/neuter,
+	"Pig" = /mob/living/basic/pig,
+	"Pug" = /mob/living/basic/pet/dog/pug,
+	"Rabbit" = /mob/living/basic/rabbit,
+	"Sloth" = /mob/living/basic/sloth,
+	"Snake" = /mob/living/basic/snake,
+	"Space carp" = /mob/living/basic/carp/pet,
+	"Space cat" = /mob/living/simple_animal/pet/cat/space,
```

Comment:
The space animals are kinda meant to be rare/unique animals, same goes with the void puppy, the cake cat and the bread cat, amd I feel like that kinda makes them lose a bit of what makes them unique/interesting, i.e. the fact that they're rare.